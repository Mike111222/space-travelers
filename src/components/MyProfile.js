import React from 'react';
import { useSelector } from 'react-redux';
import { allMissions } from '../redux/mission/missionSlice';
// import { reserve } from '../redux/rocket/rocketSlice';

const MyProfile = () => {
  const missions = useSelector(allMissions);
  const missionsArray = missions.filter((mission) => mission.joined);
  // const rocketData = reserve.filter((data) => data.joined);
  return (
    <>
      <div className="mainContainer">
        <section className="myMissions">
          <span className="sectionTitle">My Missions</span>
          {missionsArray.length > 0 ? (
            <ul className="myMissionsList">
              {missionsArray.map((mission) => {
                if (mission?.joined) {
                  return (
                    <li className="item" key={mission.id}>
                      {mission.missionName}
                    </li>
                  );
                }
                return null;
              })}
            </ul>
          ) : (
            <span className="item">No Missions Joined</span>
          )}
        </section>
      </div>
    </>
  );
};

export default MyProfile;
