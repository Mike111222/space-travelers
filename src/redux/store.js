import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './mission/missionSlice';
import rocketsReducer from './rocket/rocketSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionReducer,
  },
});

export default store;
