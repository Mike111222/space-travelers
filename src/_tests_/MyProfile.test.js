import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import MyProfile from '../components/MyProfile';

test('render correctly and matches the My Profile snapshot', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <MyProfile />
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
