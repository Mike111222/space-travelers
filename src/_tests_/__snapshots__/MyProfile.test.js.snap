// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly and matches the My Profile snapshot 1`] = `
<DocumentFragment>
  <div
    class="mainContainer"
  >
    <section
      class="myMissions"
    >
      <span
        class="sectionTitle"
      >
        My Missions
      </span>
      <span
        class="item"
      >
        No Missions Joined
      </span>
    </section>
    <div
      class="rockets"
    >
      <span
        class="sectionTitle"
      >
        My Rockets
      </span>
      <p>
        No rockets reserved yet!
      </p>
    </div>
  </div>
</DocumentFragment>
`;
